packages/xrpc-server/src/types.ts

```diff
@@ -3,7 +3,8 @@ import { isHttpError } from 'http-errors'
 import zod from 'zod'
 import { ResponseType, ResponseTypeStrings } from '@atproto/xrpc'
 
-export type Params = typeof express.request['query']
+export type UndecodedParams = typeof express.request['query']
```

Comment:
I just took a closer look at this type, and I think in the future we may want to switch query param parsers to guarantee much simpler output.  I believe there could be objects kicking around in here, which I'd advocate for nipping in the bud at the parsing step.  Poor `decodeQueryParam` can't assume anything about the value, when in actually we can do much better, guaranteeing `string | string[]`.  I could handle this in my pending xrpc-server work if we want to make that change.

Reply:
Sure whatever you think is best